MLS.Agent.Tests/AgentService.cs

```diff
@@ -41,11 +43,17 @@ public AgentService(StartupOptions options = null, IDirectoryAccessor directoryA
 
         public void Dispose() => _disposables.Dispose();
 
-        private TestServer CreateTestServer() => new TestServer(CreateWebHostBuilder());
+        private TestServer CreateTestServer()
+        {
+            var featureCollection = new FeatureCollection();
+            featureCollection.Set<IServerAddressesFeature>(new ServerAddressesFeature());
+            return new TestServer(CreateWebHostBuilder(), featureCollection);
```

Comment:
Ideally this should not be a special set up that happens in test one way and in the application's startup code in another way.

Reply:
https://github.com/dotnet/try/pull/246/files#diff-cd33a75e1de2941d5f8cbf5c1494709dL63.

The setup is using the same extension method as the main Program to configure the application url

Reply:
https://github.com/dotnet/try/pull/246/files#diff-7e4af26b8c2fa96387cebaf1ae6d7716R14

Reply:
For more context, since we are reading the url at which the application is being launched, using the ServerFeatures here: https://github.com/dotnet/try/pull/246/files#diff-ca9274b169fb9c6f45c3159115b91007R186, we need to set the feature collection in the TestServer to not to be null. https://github.com/aspnet/Hosting/pull/969